Build fails: StringChunk declared as crate-private, can't leak?

The below is seen when building against rustc and cargo 1.73.0, as shipped in the 23.11 release of nixpkgs, on MacOS/aarch64:

```
python3.12-jiter> error[E0446]: crate-private type `StringChunk` in public interface
python3.12-jiter>    --> crates/jiter/src/simd_aarch64.rs:195:1
python3.12-jiter>     |
python3.12-jiter> 195 | / pub fn decode_string_chunk(
python3.12-jiter> 196 | |     data: &[u8],
python3.12-jiter> 197 | |     mut index: usize,
python3.12-jiter> 198 | |     mut ascii_only: bool,
python3.12-jiter> 199 | |     allow_partial: bool,
python3.12-jiter> 200 | | ) -> JsonResult<(StringChunk, bool, usize)> {
python3.12-jiter>     | |___________________________________________^ can't leak crate-private type
python3.12-jiter>     |
python3.12-jiter>    ::: crates/jiter/src/string_decoder.rs:166:1
python3.12-jiter>     |
python3.12-jiter> 166 |   pub(crate) enum StringChunk {
python3.12-jiter>     |   --------------------------- `StringChunk` declared as crate-private
python3.12-jiter> error[E0446]: crate-private type `StringChunk` in public interface
python3.12-jiter>    --> crates/jiter/src/string_decoder.rs:148:1
python3.12-jiter>     |
python3.12-jiter> 148 | / pub fn decode_chunk(
python3.12-jiter> 149 | |     data: &[u8],
python3.12-jiter> 150 | |     index: usize,
python3.12-jiter> 151 | |     ascii_only: bool,
python3.12-jiter> 152 | |     allow_partial: bool,
python3.12-jiter> 153 | | ) -> JsonResult<(StringChunk, bool, usize)> {
python3.12-jiter>     | |___________________________________________^ can't leak crate-private type
python3.12-jiter> ...
python3.12-jiter> 166 |   pub(crate) enum StringChunk {
python3.12-jiter>     |   --------------------------- `StringChunk` declared as crate-private
python3.12-jiter> For more information about this error, try `rustc --explain E0446`.
python3.12-jiter> error: could not compile `jiter` (lib) due to 2 previous errors
python3.12-jiter> warning: build failed, waiting for other jobs to finish...
python3.12-jiter> 💥 maturin failed
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -9,12 +9,33 @@ on:
   pull_request: {}
 
 jobs:
+  resolve:
+
+    runs-on: ubuntu-latest
+    outputs:
+      MSRV: ${{ steps.resolve-msrv.outputs.MSRV }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: set up python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: resolve MSRV
+        id: resolve-msrv
+        run:
+          echo MSRV=`python -c 'import tomllib; print(tomllib.load(open("Cargo.toml", "rb"))["workspace"]["package"]["rust-version"])'` >> $GITHUB_OUTPUT
+
   test-linux:
+    needs: [resolve]
     name: test rust-${{ matrix.rust-version }} on linux
     strategy:
       fail-fast: false
       matrix:
         rust-version: [stable, nightly]
+        include:
+          - rust-version: ${{ needs.resolve.outputs.MSRV }}
 
     runs-on: ubuntu-latest
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,6 +15,7 @@ keywords = ["JSON", "parsing", "deserialization", "iter"]
 categories = ["parser-implementations", "parsing"]
 homepage = "https://github.com/pydantic/jiter/"
 repository = "https://github.com/pydantic/jiter/"
+rust-version = "1.75.0"
 
 [profile.bench]
 debug = true

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,7 +15,7 @@ keywords = ["JSON", "parsing", "deserialization", "iter"]
 categories = ["parser-implementations", "parsing"]
 homepage = "https://github.com/pydantic/jiter/"
 repository = "https://github.com/pydantic/jiter/"
-rust-version = "1.75.0"
+rust-version = "1.73.0"
 
 [profile.bench]
 debug = true

--- a/crates/jiter/src/simd_aarch64.rs
+++ b/crates/jiter/src/simd_aarch64.rs
@@ -192,7 +192,7 @@ const CONTROL_16: SimdVecu8_16 = simd_const!([32u8; 16]);
 const ASCII_MAX_16: SimdVecu8_16 = simd_const!([127u8; 16]);
 
 #[inline(always)]
-pub fn decode_string_chunk(
+pub(crate) fn decode_string_chunk(
     data: &[u8],
     mut index: usize,
     mut ascii_only: bool,

--- a/crates/jiter/src/string_decoder.rs
+++ b/crates/jiter/src/string_decoder.rs
@@ -146,7 +146,7 @@ fn decode_to_tape<'t, 'j>(
 }
 
 #[inline(always)]
-pub fn decode_chunk(
+pub(crate) fn decode_chunk(
     data: &[u8],
     index: usize,
     ascii_only: bool,

